zcash_client_backend/src/data_api/testing/transparent.rs

```diff
@@ -192,7 +226,14 @@ where
         .unwrap();
 
     // The wallet should detect the balance as available
-    check_balance(&st, 0, value);
+    let mut zero_or_one_conf_value = Balance::ZERO;
+
+    // add the spendable value to the expected balance
+    zero_or_one_conf_value.add_spendable_value(value).unwrap();
+
+    check_balance::<DSF>(&st, &account, taddr, 0, &zero_or_one_conf_value);
+    // we currently treat min_confirmation the same regardless they are 0 or 1
+    check_balance::<DSF>(&st, &account, taddr, 1, &zero_or_one_conf_value);
```

Comment:
It seems redundant to both make two calls to `check_balance` here (and below), and to also have the `if min_confirmations == 0 || min_confirmations == 1 { ... }` code in `check_balance`. (Even though they are currently not quite equivalent.)

Reply:
makes sense. I added some docs to the `check_balance` function instead so callers are warned that the assumption of the zero and one `min_confirmations` being treated equally will be checked